### ISSUE TITLE
Catalog performance issues

### ISSUE THREAD

**Issue by kyyberi** [apinf/platform#2351]
I've been using the catalog a lot lately and I'm somewhat worried about the performance. We have now like 120 APIs and getting listing of them takes several seconds. The same with organization views. Do we have means to collect facts about load times from end-user pov? There are free external tools for this, but do we have that info somewhere in our logs/system already?

**Comment by bajiat** [comment]
@marla-singer and @brylie Could this issue be related to 2221? Too much information is published to the catalog and hence the performance is affected?

**Comment by brylie** [comment]
While the number of fields contained in API Catalog items would decrease performance, the bigger culprit might be the number of **documents** we are requesting. Pagination is supposed to improve performance in this regard, as we only ask for one page of documents at a time. E.g. if a page is 12 documents, we only ask for those 12.

**Comment by brylie** [comment]
There could be another subscription active on the Catalog, such as **allApiLogos**. In general, #2221 is in the right 'spirit'. We need to **audit the subscriptions on the catalog page** more broadly.

**Comment by brylie** [comment]
Here is a screenshot of the active subscriptions on the API Catalog. On the right of the image are all of the **client-side collections**. The number in parenthesis is the **number of documents currently subscribed** from each collection.
![screenshot_20170323_105927](https://cloud.githubusercontent.com/assets/17307/24239824/0d936792-0fb8-11e7-97e3-b0e795a08863.png)

What this might be showing is that we are over-subscribing on the **API Logos** and perhaps **Users** collections. This would cause the page load delay, since the Catalog view waits for all subscriptions to be **ready** before rendering.

**Comment by kyyberi** [comment]
I haven't done thorough testing yet, but first problem with the sucking performance seems to be huge JS that gets loaded first. According to two different test frameworks, JS loading takes 2,7-3 seconds. A good responsive page gets loaded in a second or less. That is sort of expected nowadays. 5-10 seconds is murder. 

I know the website performance testers do not offer very reliable testbench, but perhaps someone needs to check this thing too and "database" optimation.

**Comment by bajiat** [comment]
@marla-singer Can I co-assign you to this task with @brylie? You have a lot of experience with fixing the publications.

**Comment by marla-singer** [comment]
@bajiat Sure

**Comment by marla-singer** [comment]
@bajiat @brylie Today I was working on the preference of API Catalog. Shared the results:
1. Detected the wrong behavior of `Meteor.Pagination` feature. 
2. I thought all problems came from this package then I updated package to 1.0.17 version and tested again. The subscription behavior was changed but another problem came - subscribed data was storaged in miniMongo after destroying template. It turned out @brylie opened the [same issue](https://github.com/Kurounin/Pagination/issues/35) and the package author fixed it quickly in 1.0.20 version.
3. I updated the package again to 1.0.20 version and anyway it didn't work as expected. Then I researched the changes by author in 1.0.20 version and detected the typo bug
4. Fixed it on local, tested and created the [forked PR](https://github.com/Kurounin/Pagination/pull/39)

After fixing the Pagination package I detected that it is not solved our main problem at all 😕 
The main reason is incorrect using of subscription `allApiLogo`.
For rendering API logo in API card we are using the template `viewApiLogo`. If API catalog has 20 APIs then API catalog renders 20 templates `viewApiLogo`. Each template has the logic where has the subscription `allApiLogo`. It means that subscribing on **all API logo** begins **21 times**! One is on API catalog and 20 times because of 20 templates.
Because of incorrect using of subscription for API logo we have 8 times of subscribing on all API logo on the front page. The same result is on the Organization profile with connected APIs.

The next step is to change subscribing of API logo.
